crates/core/src/dom/mutations_writer.rs

```diff
@@ -48,6 +48,17 @@ impl<'a> MutationsWriter<'a> {
                     continue;
                 }
 
+                let layer_state = node.get::<LayerState>();
+                let does_node_have_state = layer_state.is_some();
+
+                // There might exist Nodes in the RealDOM with no states yet,
+                // this is mainly due to nodes being created in the same run as when this function (remove) is being called,
+                // like nodes created by loaded templates.
+                // In this case we can safely skip these nodes.
+                if !does_node_have_state {
```

Comment:
Nitpicking: Maybe just call the variable `node_state_is_empty = layer_state.is_none()` so we don't have to think about the `!` haha.